src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/Contracts/ExecutionPlanGraph.cs

```diff
@@ -161,4 +165,24 @@ public class ExecutionPlanGraphInfo
         /// </summary>
         public string GraphFileType { get; set; }
     }
+
+    public class Badge
+    {
+        /// <summary>
+        /// Type of the node overlay. This determines the icon that is displayed for it
+        /// </summary>
+        public BadgeType Type { get; set; }
+
+        /// <summary>
+        /// Text to display for the overlay tooltip
+        /// </summary>
+        public string Tooltip { get; set; }
+    }
+
+    public enum BadgeType
+    {
+        WARNING,
```

Comment:
nit: it's a good idea to have explicit values assigned to enum values, particularly if the enum will be duplicated in different codebase.  it prevents unexpected changes when adding values/changing order in the future.